src/memoryarena.cpp

```diff
@@ -17,6 +17,9 @@
  *
  ***********************************************************************/
 
+#ifndef MEMORY_ARENA_CPP
+#define MEMORY_ARENA_CPP
```

Comment:
As the commit message for these changes notes, I ran into trouble with `src/memoryarena.cpp` when `gcc` went to compile that file _itself_. It includes `memoryarena.h`, which in turn includes `memoryarena.cpp`... Honestly, I'm confused how that even works under `qmake`, it doesn't seem like it should.

Adding include guards to the `.cpp` file itself fixed that issue for me, though.

Similarly, adding an empty `src/textblockdata.cpp` was necessary to get CMake's `AUTOMOC` to moc the corresponding header file.